x-pack/legacy/plugins/file_upload/public/util/geo_json_clean_and_validate.js

```diff
@@ -7,47 +7,59 @@
 import * as jsts from 'jsts';
 import rewind from 'geojson-rewind';
 
-export function geoJsonCleanAndValidate(parsedFile) {
-
-  // Remove bbox property pending fix of bbox parsing issue in jsts lib
-  const { bbox, ...handledGeoJsonProperties } = parsedFile; // eslint-disable-line no-unused-vars
-
-  const reader = new jsts.io.GeoJSONReader();
-  const geoJson = reader.read(handledGeoJsonProperties);
-  const isSingleFeature = parsedFile.type === 'Feature';
-  const features = isSingleFeature
-    ? [{ ...geoJson }]
-    : geoJson.features;
-
-  // Pass features for cleaning
-  const cleanedFeatures = cleanFeatures(features);
-
-  // Put clean features back in geoJson object
-  const cleanGeoJson = {
-    ...parsedFile,
-    ...(isSingleFeature
-      ? cleanedFeatures[0]
-      : { features: cleanedFeatures }
-    ),
+const GeoJSONIO = (() => {
```

Comment:
This seems an elaborate approach for a simple issue.

```
const geojsonReader = new jsts.io.GeoJSONReader();
const geojsonWriter = new jsts.io.GeoJSONWriter();
```

Apart from this simplification, imho, we no longer need the use of sub-module pattern with the self-invoking function to mimic this OO-concept of encapsulation, and I would strongly advise _against_ using such elaborate constructs in the future.

- We have actual modules now. Anything that is conceivably a module, can just be extracted in one.
- We have actual classes, which are syntactic sugar to model exactly this. Internal state local to the instance (geojsonReader, geojsonWriter), and a public API (read, write) on the prototype.

Reply:
Yeah, writing a class felt "heavy" for what's effectively just one util function we care about exporting. That's definitely not always the case though as you can see with the `PatternReader` class. This logic just allocated resources explicitly on first function call which sounded nice at the time. Having said that, I can appreciate it not being the prettiest thing to look at so happy to change/simplify it!